service/inject/api/src/main/java/io/helidon/service/inject/api/Injection.java

```diff
@@ -0,0 +1,515 @@
+     * <p>
+     * Note that instances that are not created within a scope will not have this method invoked (as their lifecycle is not
+     * fully managed by the injector).
+     */
+    @Documented
+    @Retention(RetentionPolicy.CLASS)
+    @Target(ElementType.METHOD)
+    public @interface PreDestroy {
+    }
+
+    /**
+     * Indicates the desired startup sequence for a service class. This is not used internally by Injection, but is available as a
+     * convenience to the caller in support for a specific startup sequence for service activations.
+     */
+    @Documented
+    @Retention(RetentionPolicy.CLASS)
+    @Target(ElementType.TYPE)
+    public @interface RunLevel {
+
+        /**
+         * Represents an eager singleton that should be started at "startup". Note, however, that callers control the actual
+         * activation for these services, not the framework itself, as shown below:
+         * <pre>
+         * {@code
+         * List<ServiceProvider<Object>> startupServices = services
+         *               .lookup(ServiceInfoCriteria.builder().runLevel(RunLevel.STARTUP).build());
+         *       startupServices.stream().forEach(ServiceProvider::get);
+         * }
+         * </pre>
+         */
+        int STARTUP = 10;
+
+        /**
+         * Anything > 0 is left to the underlying provider implementation's discretion for meaning; this is just a default for
+         * something that is deemed "other than startup".
+         */
+        int NORMAL = 100;
+
+        /**
+         * The service ranking applied when not declared explicitly.
+         *
+         * @return the startup int value, defaulting to {@link #NORMAL}
+         */
+        int value() default NORMAL;
+    }
+
+    /**
+     * A service that has instances created for each named instance of the service it is driven by.
+     * The instance created will have the same {@link Injection.Named} qualifier as the
+     * driving instance (in addition to all qualifiers defined on this service).
+     * <p>
+     * There are a few restrictions on this type of services:
+     * <ul>
+     * <li>The service MUST NOT implement {@link java.util.function.Supplier}</li>
+     * <li>The service MUST NOT implement {@link Injection.InjectionPointProvider}</li>
+     * <li>The service MUST NOT implement {@link Injection.ServicesProvider}</li>
+     * <li>All types that inherit from this service will also inherit the driven by</li>
+     * <li>There MAY be an injection point of the type defined in {@link #value()}, without any qualifiers -
+     * this injection point will be satisfied by the driving instance</li>
+     * <li>There MAY be a {@link String} injection point qualified with
+     * {@link Injection.DrivenByName} - this injection point will be satisfied by the
+     * name of the driving instance</li>
+     * </ul>
+     */
+    @Documented
+    @Retention(RetentionPolicy.CLASS)
+    @Target(ElementType.TYPE)
+    public @interface DrivenBy {
```

Comment:
I like `CreateFor`, so I have refactored it so. I know it uses a different style, but it seems correct to say in the source "create instances of this service for each instance of that service". I hope this will be OK with the team, as it is quite a heavy refactoring.